src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,357 @@
+    X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+
+    Igens = unique!([det(X[:, nb]) for nb in NBs ])
+
+    if length(Igens) == 0
+        return (X, SinvR, iota(ideal(R, [])))
+    else
+        Ipre = ideal(SinvR, Igens)
+        return (X, SinvR, Ipre)
+    end
+
+end
+
+@doc Markdown.doc"""
+    matroid_stratum_matrix_coordinates(M::Matroid, B::Vector{Int64}, F::AbstractAlgebra.Ring = ZZ)
+
+Return the data of the coordinate ring of the matroid stratum of M in the Grassmannian with respect to matrix coordinates. Here, ```B``` is a basis of ```M``` and the submatrix with columns indexed by ```B''' is the identity. This function returns a pair ```(SinvR, I)``` where the coordinate ring is isomorphic to ```SinvR / I```.
+
+# Examples
+```jldoctest
+julia> M = fano_matroid()
+julia> matroid_stratum_matrix_coordinates(M, [1,2,4], GF(2))
+([1 0 x[1, 1] 0 x[1, 2] 0 x[1, 4]; 0 1 x[2, 1] 0 0 x[2, 3] x[2, 4]; 0 0 0 1 x[3, 2] x[3, 3] x[3, 4]], localization of Multivariate Polynomial Ring in 9 variables x[1, 1], x[2, 1], x[1, 2], x[3, 2], ..., x[3, 4] over Galois field with characteristic 2 at the powers of gfp_mpoly[x[3, 3], x[1, 1]*x[2, 3]*x[3, 4] + x[1, 1]*x[3, 3]*x[2, 4] + x[2, 1]*x[3, 3]*x[1, 4], x[2, 3]*x[1, 4], x[1, 2]*x[2, 3]*x[3, 4] + x[1, 2]*x[3, 3]*x[2, 4] + x[3, 2]*x[2, 3]*x[1, 4], x[3, 2], x[1, 1]*x[3, 2]*x[2, 4] + x[2, 1]*x[1, 2]*x[3, 4] + x[2, 1]*x[3, 2]*x[1, 4], x[1, 2]*x[2, 4], x[2, 1], x[1, 1]*x[3, 4]], ideal in localization of Multivariate Polynomial Ring in 9 variables x[1, 1], x[2, 1], x[1, 2], x[3, 2], ..., x[3, 4] over Galois field with characteristic 2 at the powers of gfp_mpoly[x[3, 3], x[1, 1]*x[2, 3]*x[3, 4] + x[1, 1]*x[3, 3]*x[2, 4] + x[2, 1]*x[3, 3]*x[1, 4], x[2, 3]*x[1, 4], x[1, 2]*x[2, 3]*x[3, 4] + x[1, 2]*x[3, 3]*x[2, 4] + x[3, 2]*x[2, 3]*x[1, 4], x[3, 2], x[1, 1]*x[3, 2]*x[2, 4] + x[2, 1]*x[1, 2]*x[3, 4] + x[2, 1]*x[3, 2]*x[1, 4], x[1, 2]*x[2, 4], x[2, 1], x[1, 1]*x[3, 4]] generated by the elements 0//1, (x[2, 3]*x[3, 4] + x[3, 3]*x[2, 4])//1, (x[1, 2]*x[3, 4] + x[3, 2]*x[1, 4])//1, (x[1, 1]*x[2, 4] + x[2, 1]*x[1, 4])//1, (x[1, 1]*x[3, 2]*x[2, 3] + x[2, 1]*x[1, 2]*x[3, 3])//1)
+
+```
+"""
+
+
+
+function matroid_stratum_matrix_coordinates(M::Matroid, B::Vector{Int64},
+                                            F::AbstractAlgebra.Ring = ZZ)
+    d = rank(M)
+    n = length(matroid_groundset(M))
+    Bs = bases(M)
+    R, x, xdict = make_polynomial_ring(Bs,B,F)
+    return matroid_stratum_matrix_coordinates_given_ring(d, n, M, F, B, R, x, xdict)
+end
+
+    
+function realization_bases_coordinates(Bs::Vector{Vector{Int64}}, A::Vector{Int64})
+
+    d = length(Bs[1])
+    B = A[1:d]
+    c1 = A[d+1]
+
+
+    coordBases = [b for b in Bs if length(symdiff(B,b)) == 2]
+    
+    newCoords = Vector{Vector{Int64}}([])
+    
+    for b in coordBases
+        if is_subset(b, A)
+            continue
+        end
+        
+        row_b = setdiff(B,b)[1]
+        row_b = length([a for a in B if a < row_b]) + 1
```

Comment:
This is inefficient: you construct an array just to compute its length. Better:
```suggestion
        row_b = count(a -> a < row_b, B) + 1
```
or even
```suggestion
        row_b = count(<(row_b), B) + 1
```
The same applies below and also elsewhere in this file.